Shared/UI/Transitions/TransitionRouter.swift

```diff
@@ -32,26 +32,73 @@ class TransitionRouter: NSObject, UIViewControllerAnimatedTransitioning {
     }
 
     func transitionDuration(using transitionContext: UIViewControllerContextTransitioning?) -> TimeInterval {
-        return self.fromVC.sendingPresentationType.duration
+        let isPresenting = self.operation == .push
+        if isPresenting {
+            return self.toVC.presentationType.duration
+        } else {
+            return self.fromVC.dismissalType.duration
+        }
     }
 
     func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
-        let fromTransition = self.fromVC.getTransitionType(for: self.operation, isFromVC: true)
-        let toTransition = self.toVC.getTransitionType(for: self.operation, isFromVC: false)
-
-        switch (fromTransition, toTransition) {
-            #if IOS
-        case (let .message(fromView), let .message(toView)):
-            self.messageTranstion(fromView: fromView, toView: toView, transitionContext: transitionContext)
-            #endif 
-        case (let .move(fromView), let .move(toView)):
-            self.moveTranstion(fromView: fromView, toView: toView, transitionContext: transitionContext)
-        case (let .fill(expandingView), .fade):
+        let isPresenting = self.operation == .push
+
+        let presentedVCTransition: TransitionType
+        let presentingVCTransition: TransitionType
+        if isPresenting {
+            presentedVCTransition = self.toVC.presentationType
+            presentingVCTransition = self.fromVC.getFromVCPresentationType(for: presentedVCTransition)
+        } else {
+            presentedVCTransition = self.fromVC.dismissalType
+            presentingVCTransition = self.toVC.getToVCDismissalType(for: presentedVCTransition)
+        }
+
+
+        switch presentedVCTransition {
```

Comment:
We now give priority to the presented VCs transition style. If the style requires that the presenter to be a certain type, we check for that.